spec/controllers/hyrax/dashboard/collections_controller_spec.rb

```diff
@@ -21,97 +21,133 @@
   let(:unowned_asset)  { create(:work, user: other) }
 
   let(:collection_attrs) do
-    { title: ['My First Collection'], description: ["The Description\r\n\r\nand more"], collection_type_gid: [collection_type_gid] }
+    {
+      title: ['My First Collection'],
+      description: ["The Description\r\n\r\nand more"],
+      collection_type_gid: [collection_type_gid]
+    }
   end
 
+  let(:listener) { Hyrax::Specs::SpyListener.new }
+
+  before { Hyrax.publisher.subscribe(listener) }
+  after  { Hyrax.publisher.unsubscribe(listener) }
+
   describe '#new' do
     before { sign_in user }
 
     it 'assigns @collection' do
       get :new
-      expect(assigns(:collection)).to be_kind_of(Collection)
+      expect(assigns(:collection)).to be_kind_of(Hyrax::PcdmCollection)
     end
   end
 
   describe '#create' do
     before { sign_in user }
 
-    # rubocop:disable RSpec/ExampleLength
-    it "creates a Collection" do
```

Comment:
This test is currently commented out starting around line 73.  This is related to the question about deprecated old style parameters and whether or not we can remove support for them.

Additional question about `:create` supporting passing in parameters not defined on the form (e.g. visibility).  Should this be allowed?  What is the use case for this?  For the visibility params specifically, you can set it in the edit form, just not the new form.  The question applies in general.

Reply:
Again, the only use case that I can think of that might require passing in parameters that aren't part of the form is if a collection creation action doesn't use the form - an import or something along those lines.